stumpy/fastmath.py

```diff
@@ -0,0 +1,51 @@
+import importlib
+
+from stumpy import config
+
+
+def _set(module_name, func_name, flag):
+    """
+    Set fastmath flag for a given function
+
+    Parameters
+    ----------
+    module_name : str
+        The module name
+    func_name : str
+        The function name
+    flag : set or bool
+        The fastmath flag
+
+    Returns
+    -------
+    None
+    """
+    module = importlib.import_module(f".{module_name}", package="stumpy")
+    func = getattr(module, func_name)
+    func.targetoptions["fastmath"] = flag
+    func.recompile()
+
+    return
+
+
+def _reset(module_name, func_name):
```

Comment:
I have a suggestion for the test:

First add a dummy function to `fastmath.py`:

```
@njit(fastmath=True)
def add_assoc(x, y):
    # Example taken from https://numba.pydata.org/numba-doc/dev/user/performance-tips.html#fastmath
    return (x - y) + y
```

Then, you can write your unit tests to `_set`/`_reset` this dummy function by manipulating the `fastmath` flag and asserting the results according to:

```
    # print(njit(fastmath=False)(add_assoc)(0, np.inf)) # nan
    # print(njit(fastmath=True) (add_assoc)(0, np.inf)) # 0.0
    # print(njit(fastmath={'reassoc', 'nsz'})(add_assoc)(0, np.inf)) # 0.0
    # print(njit(fastmath={'reassoc'})       (add_assoc)(0, np.inf)) # nan
    # print(njit(fastmath={'nsz'})           (add_assoc)(0, np.inf)) # nan
```